debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerStreamingChangeEventSource.java

```diff
@@ -89,15 +89,15 @@ public void execute(ChangeEventSourceContext context) throws InterruptedExceptio
                     continue;
                 }
                 // There is no change in the database
-                if (currentMaxLsn.equals(lastProcessedLsn)) {
+                if (currentMaxLsn.equals(lastProcessedPosition.getCommitLsn())) {
```

Comment:
There is a minor issue here, which may occur only for slowly changing tables. Correct me if I am wrong.

Let's consider the following scenario. In CDC table there are entries:
commitLsn / changeLsn
B   1
B   2
B   3
B   4

The connector has processed `B1` and `B2`, then crashed. The last offset persisted in Kafka was: `B2`.
The connector has been restarted.
`LastProcessedPositionOnStart = B2`
`currentMaxLsn=B`
This `if` statement prevents from processing `B3` and `B4` until the commitLsn increases to `C`.